src/app/video-player/video-player.service.ts

```diff
@@ -371,22 +371,22 @@ export class VideoPlayerService {
             this.onWatchStatusChanges.pipe(
                 tap(episode => {
                     const episodeOfBangumi = this._bangumi.episodes.find(e => e.id === this._episode.id);
-                    episodeOfBangumi.watch_progress = episode.watch_progress;
+                    episodeOfBangumi.watchProgress = episode.watchProgress;
                 }),
                 filter(episode => {
-                    return episode.watch_progress.watch_status === WatchProgress.WATCHED
+                    return episode.watchProgress.watchStatus === WatchProgress.WATCHED
                         && this._bangumi.favorite_status !== WatchProgress.WATCHED;
                 }))
                 .subscribe(() => {
                     const allWatched = this._bangumi.episodes
                         .every(episode => {
-                            return episode.watch_progress
-                                && episode.watch_progress.watch_status === WatchProgress.WATCHED;
+                            return episode.watchProgress
+                                && episode.watchProgress.watchStatus === WatchProgress.WATCHED;
                         });
 
                     if (allWatched) {
-                        this._bangumi.favorite_status = Bangumi.WATCHED;
-                        this._bangumiFavoriteChanges.next(Object.assign({}, this._bangumi));
+                        // this._bangumi.favorite_status = Bangumi.WATCHED;
```

Comment:
Why are those two lines commented out?

Reply:
I forgot to add back the logic to emit bangumi favorite status change event.
I have added it back.